contracts/zero-ex/contracts/src/features/OtcOrdersFeature.sol

```diff
@@ -326,57 +481,34 @@ contract OtcOrdersFeature is
             ));
         }
 
-        if (wethOptions == WethOptions.WrapEth) {
-            require(
-                order.takerToken == WETH,
-                "OtcOrdersFeature/INVALID_WRAP_ETH"
-            );
-            // Wrap ETH
-            WETH.deposit{value: takerTokenFilledAmount}();
-            // Transfer WETH to maker
-            WETH.transfer(order.maker, takerTokenFilledAmount);
-            if (takerTokenFilledAmount < msg.value) {
-                // Refund unused ETH
-                _transferEth(
-                    msg.sender,
-                    msg.value - uint256(takerTokenFilledAmount)
-                );
+        if (payer == address(this)) {
+            if (address(order.takerToken) == ETH_TOKEN_ADDRESS) {
+                // Transfer ETH to the maker.
+                payable(order.maker).transfer(takerTokenFilledAmount);
+            } else {
+                // Transfer this -> maker.
+                _transferERC20Tokens(
```

Comment:
when would we hit this case (transfer ERC20 tokens from a balance in address(this))

Reply:
ah that'll happen when `fillOtcOrderWithEth` is called and `makerToken` = WETH. Any other case?

Reply:
@alexkroeger multiplex I'm guessing